infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureDetector.java

```diff
@@ -239,12 +201,17 @@ private long modifyIterationTimeouts(Map<String, IClientRouter> router, PollRepo
      * @param allConnectedNodes all connected nodes
      */
     private void tunePollReportTimeouts(
-            Map<String, IClientRouter> clientRouters, Set<String> failedNodes, Set<String> allConnectedNodes) {
-        period = Math.max(initPeriodDuration, period - periodDelta);
-        tuneRoutersResponseTimeout(clientRouters, allConnectedNodes, period);
+            Map<String, IClientRouter> clientRouters, Set<String> failedNodes,
+            Set<String> allConnectedNodes) {
+
+        networkStretcher.modifyDecreasedPeriod();
+        tuneRoutersResponseTimeout(
+                clientRouters, allConnectedNodes, networkStretcher.getDecreasedPeriod()
```

Comment:
You are decreasing the timeout twice here.
modifyDecreasedPeriod already decreases it once.

Reply:
actually no, `getDecreaseTimeout` doesn't decrease the current period

Reply:
so I modify it only ones

Reply:
I get that, but you are tuning the routers with a timeout that has been decreased twice.

Reply:
fixed